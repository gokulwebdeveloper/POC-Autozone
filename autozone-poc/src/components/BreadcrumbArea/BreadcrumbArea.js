import React from 'react';
import BgImg from '../../utlis/img/bg-img/breadcumb.jpg';
import PropTypes from 'prop-types';

export default function BreadcrumbArea({title}) {
    const bgImgStyle = {
        'backgroundImage': `url(${BgImg})`
    }    
    return (
    <div className="breadcumb_area bg-img" style={bgImgStyle}>
        <div className="container h-100">
            <div className="row h-100 align-items-center">
                <div className="col-12">
                    <div className="page-title text-center">
                    <h2>{title}</h2>
                    </div>
                </div>
            </div>
        </div>
    </div>
);
}

BreadcrumbArea.propTypes = {
    title: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.number])
}

BreadcrumbArea.defaultProps = {
    title: 'title'
}  
