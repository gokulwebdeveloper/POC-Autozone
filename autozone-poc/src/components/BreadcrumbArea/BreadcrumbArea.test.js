import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import BreadcrumbArea from "./BreadcrumbArea";

let container = null;

//mount component for testing
beforeEach(() => {
  container = mount(<BreadcrumbArea title="checkout" />);
});

//Cleanup after test execution
afterEach(() => {
  container.unmount();
});

describe("test BreadCrumbArea component", () => {
  //test if component render correctly
  it("component render correctly", () => {
    expect(container.exists()).toBeTruthy();
  });

  //test if it have correct props
  it("should have title as props", () => {
    expect(container.prop("title")).toEqual("checkout");
  });

  //test if it should have one h2 for title
  it("should have one h2 for title", () => {
    expect(container.find("h2")).toHaveLength(1);
  });

  //SnapShot Match test
  it("SnapShot Match", () => {
    const container = renderer.create(<BreadcrumbArea />).toJSON();
    expect(container).toMatchSnapshot();
  });
});
