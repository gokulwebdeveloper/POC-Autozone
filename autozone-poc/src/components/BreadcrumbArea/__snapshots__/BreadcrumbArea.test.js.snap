// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test BreadCrumbArea component SnapShot Match 1`] = `
<div
  className="breadcumb_area bg-img"
  style={
    Object {
      "backgroundImage": "url(breadcumb.jpg)",
    }
  }
>
  <div
    className="container h-100"
  >
    <div
      className="row h-100 align-items-center"
    >
      <div
        className="col-12"
      >
        <div
          className="page-title text-center"
        >
          <h2>
            title
          </h2>
        </div>
      </div>
    </div>
  </div>
</div>
`;
